src/main/scala/com/cloudant/clouseau/IndexService.scala

```diff
@@ -219,7 +219,7 @@ class IndexService(ctx: ServiceContext[IndexServiceArgs]) extends Service(ctx) w
       case e: IOException => warn("Error while closing reader", e)
     }
     try {
-      ctx.args.writer.rollback()
+      ctx.args.writer.close()
```

Comment:
If I understand [the fine manual](https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/index/IndexWriter.html#rollback()) correctly, `rollback()` does the following:

1. Closes the IndexWriter without committing any changes that have occurred since the last commit
2. Removes any temporary files that had been created

And [`close()`](https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/index/IndexWriter.html#close()):

1. Commits all changes to an index
2. Waits for pending merges to complete (maybe not applicable?)
3. Closes all associated files

And there's also `close(boolean waitForMerges)` which I'm not sure applies to us since it "only meaningful when using a MergeScheduler that runs merges in background threads".

So it sounds like a semantic change is that we'd also be committing any additional changes since last commit by using `close()`, correct?

Reply:
That is what @rnewson wrote on Slack: _"close will wait for segment merges, which is a good thing. we changed another rollback case to close, when the lru evicts an index, as we were seeing indexes with huge numbers of segments, because the merges were never allowed time to finish"_.

Reply:
hm, a commit would be a bit much (in later versions of Lucene you can configure whether close() does a commit or not). I chose rollback() originally for the reasons in the doc, the quickest way to close resources without doing anything expensive.

committing the pending changes isn't _wrong_. the changes are correct, and they'd just be done identically on the next attempt. We might be wasting resources though, as the update seq is advanced separately. Unless we update the update seq in the IndexCommit we'll end up re-doing the work inbetween.

(In nouveau I changed how this is done, so that every update or delete includes its update sequence, so the JVM side is free to commit unflushed changes whenever it likes. We might change dreyfus/clouseau to do the same).